polkadot/node/collation-generation/src/tests.rs

```diff
@@ -842,236 +226,185 @@ fn distribute_collation_for_occupied_core_pre_async_backing() {
 #[rstest]
 #[case(0)]
 #[case(1)]
 #[case(2)]
-fn distribute_collation_for_occupied_cores_with_async_backing_enabled_and_elastic_scaling(
-	#[case] candidates_pending_avail: u32,
-) {
+#[case(3)]
+fn distribute_collation_with_elastic_scaling(#[case] total_cores: u32) {
 	let activated_hash: Hash = [1; 32].into();
 	let para_id = ParaId::from(5);
-	// Using latest runtime with the fancy claim queue exposed.
-	let runtime_version = RuntimeApiRequest::CLAIM_QUEUE_RUNTIME_REQUIREMENT;
-
-	let cores = (0..3)
-		.into_iter()
-		.map(|idx| {
-			CoreState::Occupied(polkadot_primitives::vstaging::OccupiedCore {
-				next_up_on_available: Some(ScheduledCore { para_id, collator: None }),
-				occupied_since: 0,
-				time_out_at: 10,
-				next_up_on_time_out: Some(ScheduledCore { para_id, collator: None }),
-				availability: Default::default(), // doesn't matter
-				group_responsible: polkadot_primitives::GroupIndex(idx as u32),
-				candidate_hash: Default::default(),
-				candidate_descriptor: dummy_candidate_descriptor_v2(dummy_hash()),
-			})
-		})
-		.collect::<Vec<_>>();
 
-	let pending_availability = (0..candidates_pending_avail)
+	let claim_queue = (0..total_cores)
 		.into_iter()
-		.map(|_idx| dummy_candidate_pending_availability(para_id, activated_hash, 0))
-		.collect::<Vec<_>>();
-
-	let claim_queue = cores
-		.iter()
-		.enumerate()
-		.map(|(idx, _core)| (CoreIndex::from(idx as u32), VecDeque::from([para_id])))
+		.map(|idx| (CoreIndex(idx), VecDeque::from([para_id])))
 		.collect::<BTreeMap<_, _>>();
-	let total_cores = cores.len();
 
 	test_harness(|mut virtual_overseer| async move {
 		helpers::initialize_collator(&mut virtual_overseer, para_id).await;
 		helpers::activate_new_head(&mut virtual_overseer, activated_hash).await;
 		helpers::handle_runtime_calls_on_new_head_activation(
 			&mut virtual_overseer,
 			activated_hash,
-			AsyncBackingParams { max_candidate_depth: 1, allowed_ancestry_len: 1 },
-			cores,
-			runtime_version,
 			claim_queue,
+			NodeFeatures::EMPTY,
 		)
 		.await;
 
 		helpers::handle_cores_processing_for_a_leaf(
 			&mut virtual_overseer,
 			activated_hash,
 			para_id,
-			// if at least 1 cores is occupied => `OccupiedCoreAssumption` is `Included`
-			// else assumption is `Free`.
-			if candidates_pending_avail > 0 {
-				OccupiedCoreAssumption::Included
-			} else {
-				OccupiedCoreAssumption::Free
-			},
-			total_cores,
-			pending_availability,
-			runtime_version,
+			(0..total_cores).collect(),
 		)
 		.await;
 
 		virtual_overseer
 	});
 }
 
-// There are variable number of cores of cores in `Free` state and async backing is enabled.
-// On new head activation `CollationGeneration` should produce and distribute a new collation
-// with proper assumption about the para candidate chain availability at next block.
 #[rstest]
-#[case(0)]
-#[case(1)]
-#[case(2)]
-fn distribute_collation_for_free_cores_with_async_backing_enabled_and_elastic_scaling(
```

Comment:
Yes. Since we no longer care about occupied cores, this test and `distribute_collation_for_occupied_cores_with_async_backing_enabled_and_elastic_scaling` have been merged into `distribute_collation_with_elastic_scaling`